File: SjoslagetFrontend/lib/model/booking_dashboard_item.dart
import 'package:frontend_shared/util.dart' show DurationFormatter;

import 'json_field.dart';

class BookingDashboardItem {
  Duration _sinceUpdated;

  final String id;
  final String reference;
  final String subCruise;
  final String firstName;
  final String lastName;
  final DateTime created;
  final DateTime updated;
  final int numberOfCabins;
  final int numberOfPax;

  BookingDashboardItem(this.id, this.reference, this.subCruise, this.firstName, this.lastName, this.created, this.updated,
      this.numberOfCabins, this.numberOfPax);

  factory BookingDashboardItem.fromMap(Map<String, dynamic> json) => BookingDashboardItem(
      json[ID],
      json[REFERENCE],
      json[SUBCRUISE],
      json[FIRSTNAME],
      json[LASTNAME],
      DateTime.parse(json[CREATED]),
      DateTime.parse(json[UPDATED]),
      json[NUMBER_OF_CABINS],
      json[NUMBER_OF_PAX]);

  String get sinceUpdated {
    if (null == _sinceUpdated) return '';

    return DurationFormatter.formatCompact(_sinceUpdated);
  }

  void update(DateTime now) {
    _sinceUpdated = now.difference(updated);
  }
}

File: SjoslagetBackend/Models/BookingDashboardItem.cs
﻿using System;

namespace Accidis.Sjoslaget.WebService.Models
{
	public sealed class BookingDashboardItem
	{
		public Guid Id { get; set; }
		public string Reference { get; set; }
		public string SubCruise { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public DateTime Created { get; set; }
		public DateTime Updated { get; set; }
		public int NumberOfCabins { get; set; }
		public int NumberOfPax { get; set; }
	}
}
